api/http.go

```diff
@@ -70,8 +75,8 @@ func (handler *hTTPHandler) ServeHTTP(w http.ResponseWriter, req *http.Request)
 		w.WriteHeader(http.StatusMethodNotAllowed)
 		return
 	}
-
-	if err := handler.msgHandler.HandlePOSTReq(ctx, req.Body,
+	body := io.LimitReader(req.Body, _maxRequestContentLength)
```

Comment:
no need to check error? i mean if it has > 5MB data, `body` will be incomplete, would that be an issue for the `HandlePOSTReq` call below?

Reply:
should be here  https://github.com/ethereum/go-ethereum/blob/master/rpc/http.go#L343
```golang
	if r.ContentLength > maxRequestContentLength {
		err := fmt.Errorf("content length too large (%d>%d)", r.ContentLength, maxRequestContentLength)
		return http.StatusRequestEntityTooLarge, err
	}
```

Reply:
so we need a similar check to validate the request?